pkg/standalone-utils/test/foundry/EclpLPOracle.t.sol

```diff
@@ -0,0 +1,648 @@
+        answers[0] = bound(answersRaw[0], uint256(_paramsAlpha), uint256(_paramsBeta));
+        answers[1] = 1e18;
+
+        IGyroECLPPool pool = createAndInitPool(_tokens, poolInitAmounts);
+        (EclpLPOracleMock oracle, AggregatorV3Interface[] memory feeds) = deployOracle(pool);
+
+        for (uint256 i = 0; i < NUM_TOKENS; i++) {
+            FeedMock(address(feeds[i])).setLastRoundData(answers[i], updateTimestamps[i]);
+        }
+
+        uint256 expectedTVL = _computeExpectedTVLBinarySearch(GyroECLPPool(address(pool)), answers);
+
+        (
+            uint80 roundId,
+            int256 lpPrice,
+            uint256 startedAt,
+            uint256 returnedUpdateTimestamp,
+            uint80 answeredInRound
+        ) = oracle.latestRoundData();
+
+        assertEq(uint256(roundId), 0, "Round ID does not match");
+        // Error tolerance of 0.1%.
+        assertApproxEqRel(
+            uint256(lpPrice),
+            expectedTVL.divUp(IERC20(address(pool)).totalSupply()),
+            1e15,
+            "LP price does not match"
+        );
+        assertEq(startedAt, 0, "Started at does not match");
+        assertEq(returnedUpdateTimestamp, minUpdateTimestamp, "Update timestamp does not match");
+        assertEq(answeredInRound, 0, "Answered in round does not match");
+    }
+
+    function _computeExpectedTVLBinarySearch(
+        GyroECLPPool pool,
+        uint256[] memory oraclePricesScaled18
+    ) private returns (uint256 expectedTVL) {
+        uint256 snapshotId = vm.snapshot();
+        uint256[] memory marketPriceBalances = _findBalancesForPrices(
+            GyroECLPPool(address(pool)),
+            oraclePricesScaled18[0].divDown(oraclePricesScaled18[1])
+        );
+        vm.revertTo(snapshotId);
+
+        return
+            marketPriceBalances[0].mulDown(oraclePricesScaled18[0]) +
+            marketPriceBalances[1].mulDown(oraclePricesScaled18[1]);
+    }
+
+    function _findBalancesForPrices(
+        GyroECLPPool pool,
+        uint256 oraclePrice
+    ) private returns (uint256[] memory marketPriceBalances) {
+        IERC20[] memory tokens;
+        uint256[] memory balancesRaw;
+        (tokens, , balancesRaw, marketPriceBalances) = vault.getPoolTokenInfo(address(pool));
+
+        (IGyroECLPPool.EclpParams memory eclpParams, IGyroECLPPool.DerivedEclpParams memory derivedEclpParams) = pool
+            .getECLPParams();
+
+        uint256 limitTokenABalance = uint256(0);
+        uint256 limitTokenBBalance = uint256(0);
+
+        for (uint256 i = 0; i < 255; i++) {
+            (int256 a, int256 b) = _computeOffsetFromBalances(marketPriceBalances, eclpParams, derivedEclpParams);
+            uint256 price = _computePrice(marketPriceBalances, eclpParams, a, b);
+
+            if (
+                (price > oraclePrice && (price - oraclePrice).divDown(oraclePrice) < 1e6) ||
+                (price < oraclePrice && (oraclePrice - price).divDown(oraclePrice) < 1e6)
```

Comment:
This is essentially `|price - oraclePrice| / oraclePrice` right? 
There might be a math helper to get the absolute value of the difference directly; it could improve readability a bit.